scrapers_library/data_portals/ckan/README.md

```diff
@@ -0,0 +1,215 @@
+# CKAN Scraper
+
+## Introduction
+
+This scraper can be used to retrieve package information from [CKAN](https://ckan.org/), which hosts open data projects such as <https://data.gov/>. CKAN API documentation can be found at <https://docs.ckan.org/en/2.9/api/>.
+
+Running the scraper will output a list of packages to a CSV file using the search terms.
+
+## Definitions
+
+* `Package` - Also called a dataset, is a page containing relevant information about a dataset. For example, this page is a package: <https://catalog.data.gov/dataset/electric-vehicle-population-data>.
+* `Collection` - A grouping of child packages, related to a parent package. This is seperate from a group.
+* `Group` - Also called a topic, is a grouping of packages. Packages in a group do not have a parent package. Groups can also contain subgroups.
+* `Organization` - Organizations are what the data in packages belong to, such as "City of Austin" or "Department of Energy". Organization types are groups of organizations that share something in common with each other.
+
+## Files
+
+* `scrape_ckan_data_portals.py` - The main scraper file. Running this will execute a search accross multiple CKAN instances and output the results to a CSV file.
+* `search_terms.py` - The search terms and CKAN portals to search from.
+* `ckan_scraper_toolkit.py` - Toolkit of functions that use ckanapi to retrieve packages from CKAN data portals.
+
+## Setup
+
+1. In a terminal, navigate to the CKAN scraper folder
+    ```cmd
+    cd scrapers_library/data_portals/ckan/
+    ```
+2. Create a Python virtual environment
+    ```cmd
+    python -m venv venv
+    ```
+3. Install the requirements
+    ```cmd
+    pip install -r requirements.txt
+    ```
+4. Run the multi-portal CKAN scraper
+    ```cmd
+    python scrape_ckan_data_portals.py
+    ```
+
+## How can I tell if a website I want to scrape is hosted using CKAN?
+
+There's no easy way to tell, some websites will reference CKAN or link back to the CKAN documentation while others will not. There doesn't seem to be a database of all CKAN instances either.
+
```

Comment:
Again, excellent, very thorough description of how to approach all of this. The README overall is very well done.